debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/connection/PostgresConnectionIT.java

```diff
@@ -99,4 +110,70 @@ public void shouldDropReplicationSlot() throws Exception {
             assertTrue(connection.dropReplicationSlot("test"));
         }
     }
+
+    @Test
+    public void shouldDetectRunningConncurrentTxOnInit() throws Exception {
+        Testing.Print.enable();
+        // drop the slot from the previous connection
+        final String slotName = "block";
+        try (PostgresConnection connection = TestHelper.create()) {
+            // try to drop the previous slot
+            connection.dropReplicationSlot(slotName);
+            connection.execute(
+                    "DROP SCHEMA IF EXISTS public CASCADE",
+                    "CREATE SCHEMA public",
+                    "CREATE TABLE test(pk serial, PRIMARY KEY (pk))"
+            );
+        }
+
+        try (PostgresConnection blockingConnection = TestHelper.create("blocker")) {
+            // Create an unfinished TX
+            blockingConnection.connection().setTransactionIsolation(Connection.TRANSACTION_SERIALIZABLE);
+            blockingConnection.connection().setAutoCommit(false);
+            blockingConnection.executeWithoutCommitting(
+                    "INSERT INTO test VALUES(default)"
+            );
+            Testing.print("Blocking exception started");
+
+            final Future<?> f1 = Executors.newSingleThreadExecutor().submit(() -> {
+                // Create a replication connection that is blocked till the concurrent TX is completed
+                try (ReplicationConnection replConnection = TestHelper.createForReplication(slotName, false)) {
+                    Testing.print("Connecting with replication connection 1");
+                    assertTrue(replConnection.isConnected());
+                    Testing.print("Replication connection 1 - completed");
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            Thread.sleep(3000);
+
+            final Future<?> f2 = Executors.newSingleThreadExecutor().submit(() -> {
+                // Create a replication connection that receives confirmed_flush_lsn == null
+                try (ReplicationConnection replConnection = TestHelper.createForReplication(slotName, false)) {
+                    Testing.print("Connecting with replication connection 2");
+                    assertTrue(replConnection.isConnected());
+                    Testing.print("Replication connection 2 - completed");
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            Thread.sleep(3000);
+            blockingConnection.connection().commit();
+            Testing.print("Blocking exception finished");
+
+            Thread.sleep(6000);
```

Comment:
Whenever you do this, a little kitten painfully dies ;)

In more seriousness, I'm concerned about the increased duration of test execution by things like these. It appears as if we should begin to divide the suite into categories, so to have a way to skip such long-running tests.